subsys/pcd/src/pcd.c

```diff
@@ -229,7 +231,9 @@ static int network_core_pcd_cmdset(uint32_t cmd, const void *src_addr, size_t le
 #ifdef CONFIG_PCD_READ_NETCORE_APP_VERSION
 int pcd_network_core_app_version(uint8_t *buf, size_t len)
 {
-	if (buf == NULL || len < 4) {
+	if (buf == NULL || len < 4
+	   || ((uintptr_t)buf < (uintptr_t)&_image_ram_start)
+	   || ((uintptr_t)buf > (uintptr_t)&_image_ram_end)) {
```

Comment:
Where should this be placed, because this will exclude the 0x2000 bytes at the start of RAM which are set up when PCD is enabled?

Reply:
This buffer can be anywhere in RAM as long as it's in RAM and not in FLASH. 

The idea here is that you will put the buffer in the stack as a variable `uint32_t`  or `uint8_t arr[4]`or a static allocated RAM buffer/variable and both those I would assume is limited by Zephyr's build system to be within `_image_ram_start` and `_image_ram_end`.

I could limit it with `PM_PCD_SRAM_ADDRESS`  and `PM_RPMSG_NRF53_SRAM_END_ADDRESS` but it feels a bit wrong as the linker wouldn't place any of the above mentioned outside the range of `_image_ram_start` and `_image_ram_end`.